src/webots/gui/WbMainWindow.cpp

```diff
@@ -1619,11 +1632,134 @@ void WbMainWindow::exportHtml() {
+      map["name"] = "textures[]";
+    }
+
+    mainPart.setHeader(QNetworkRequest::ContentDispositionHeader,
+                       QVariant("form-data; name=" + map["name"] + "; filename=" + filename));
+
+    // read file content
+    if (filename != "") {
+      QFile *file = new QFile(map["foldername"] + filename);
+      file->open(QIODevice::ReadOnly);
+      mainPart.setBodyDevice(file);
+      file->setParent(multiPart);
+    }
+    multiPart->append(mainPart);
+  }
+  // add other information
+  QMap<QString, QString> uploadInfo;
+  uploadInfo["type"] = type;
+  uploadInfo["user"] = "null";
+  uploadInfo["password"] = "null";
+
+  QMapIterator<QString, QString> iteratorUploadInfo(uploadInfo);
+  while (iteratorUploadInfo.hasNext()) {
+    iteratorUploadInfo.next();
+    QHttpPart infoPart;
+    infoPart.setHeader(QNetworkRequest::ContentDispositionHeader, QVariant("form-data; name=" + iteratorUploadInfo.key()));
+    infoPart.setBody(iteratorUploadInfo.value().toUtf8());
+    multiPart->append(infoPart);
+  }
+
+  QNetworkReply *reply = WbNetwork::instance()->networkAccessManager()->post(request, multiPart);
+
+  mUploadProgressDialog = new QProgressDialog(tr("Uploading on %1...").arg(uploadUrl), "Cancel", 0, 100, this);
+  mUploadProgressDialog->setWindowTitle(tr("%1").arg(uploadUrl));
+  mUploadProgressDialog->show();
+  connect(reply, &QNetworkReply::uploadProgress, this, &WbMainWindow::updateUploadProgressBar);
+
+  multiPart->setParent(reply);
+  connect(reply, &QNetworkReply::finished, this, &WbMainWindow::uploadFinished, Qt::UniqueConnection);
+}
+
+void WbMainWindow::updateUploadProgressBar(qint64 bytesSent, qint64 bytesTotal) {
+  if (bytesTotal > 0)
+    mUploadProgressDialog->setValue(((double)bytesSent / (double)bytesTotal) * 100.0);
+}
+
+void WbMainWindow::uploadFinished() {
+  QNetworkReply *reply = dynamic_cast<QNetworkReply *>(sender());
+  assert(reply);
+  if (!reply)
+    return;
+
+  disconnect(reply, &QNetworkReply::uploadProgress, this, &WbMainWindow::updateUploadProgressBar);
+  disconnect(reply, &QNetworkReply::finished, this, &WbMainWindow::uploadFinished);
+
+  const QStringList answers = QString(reply->readAll().data()).split("\n");
+  QString url;
+
+  foreach (const QString &answer, answers) {
+    if (answer.startsWith('{')) {  // we get only the json, ignoring the possible warnings
+      QJsonDocument document = QJsonDocument::fromJson(answer.toUtf8());
+      QJsonObject jsonAnswer = document.object();
+      url = jsonAnswer["url"].toString();
+    }
+  }
+  if (url.isEmpty()) {
+    mUploadProgressDialog->close();
+    QString error = reply->error() ? reply->errorString() : "No server answer.";
+    WbMessageBox::critical(tr("Upload failed. Error::%1").arg(error), this, tr("Webots.cloud"));
+  } else {
+    WbLog::info(tr("link: %1\n").arg(url));
+
+    mLinkWindow = new WbLinkWindow(this);
+    mLinkWindow->setLabelLink(url);
+    mLinkWindow->exec();
+  }
```

Comment:
There is a memory leak here which is the cause of the tmp files not being deleted on Windows:
```suggestion
  }
  delete reply;
```